sync.cmd

```diff
@@ -0,0 +1,74 @@
+@echo off
+setlocal
+
+set synclog=sync.log
+echo Running Sync.cmd %* > %synclog%
+
+if [%1]==[] (
+  set src=true
+  set packages=true
+  goto Begin
+)
+
+set src=false
+set packages=false
+
+:Loop
+if [%1]==[] goto Begin
+
+if /I [%1]==[/?] goto Usage
+
+if /I [%1] == [/p] (
+    set packages=true
+    goto Next
+)
+
+if /I [%1] == [/s] (
+    set src=true
+    goto Next
+)
+
+echo Unrecognized argument '%1'
+goto Usage
+
+:Next
+shift /1
+goto Loop
+
+:Begin
+
+call %~dp0init-tools.cmd
+
+if [%src%] == [true] (
+  echo Fetching git database from remote repos ...
+  call git fetch --all -p >> %synclog%
+  if NOT [%ERRORLEVEL%]==[0] (
+    echo ERROR: An error occurred while fetching remote source code, see %synclog% for more details.
+    exit /b
+  )
+)
+
+if [%packages%] == [true] (
+  echo Restoring all packages ...
+  echo msbuild %~dp0build.proj /t:BatchRestorePackages /nologo /v:minimal /p:RestoreDuringBuild=true /flp:v=diag;Append;LogFile=%synclog% >> %synclog%
+  call msbuild.exe %~dp0build.proj /t:BatchRestorePackages /nologo /v:minimal /p:RestoreDuringBuild=true /flp:v=diag;Append;LogFile=%synclog%
+  if NOT [%ERRORLEVEL%]==[0] (
+    echo ERROR: An error occurred while syncing packages, see %synclog% for more details. There may have been networking problems so please try again in a few minutes.
+    exit /b
+  )
+)
+
+echo Done Syncing.
+exit /b 0
+
+goto :EOF
+
+:Usage
+echo.
+echo Repository syncing script.
+echo.
+echo Options:
+echo     /s     - Fetches source history from all configured remotes (git fetch --all)
```

Comment:
Would it be useful to also fast-forward origin to upstream and local to origin?  That's what the internal git-get script does, and that's what I proposed for the Sync script (see [Linux workflow proposal](https://microsoft.sharepoint.com/teams/netfx/corefx/_layouts/OneNote.aspx?id=%2Fteams%2Fnetfx%2Fcorefx%2FDocuments%2FCoreFx%20Notes&wd=target%28Engineering%2FEngineering.one%7C70363CF0-E56F-4207-9066-875EB8C74AB0%2FLinux%20workflow%20proposal%7CDA704B43-BE13-458F-AFE9-BA99F6612220%2F%29))  I also proposed an option to rebase changes from master to the current local branch.

Perhaps those can be incremental additions which would come later, though.